compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -64,6 +64,47 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
 
                 rhs_then_block.unit()
             }
+            ExprKind::LogicalOp { op: LogicalOp::Or, lhs, rhs } => {
+                let local_scope = this.local_scope();
+                let (lhs_success_block, failure_block) =
+                    this.in_if_then_scope(local_scope, expr_span, |this| {
+                        this.then_else_break(
+                            block,
+                            &this.thir[lhs],
+                            temp_scope_override,
+                            local_scope,
+                            variable_source_info,
+                        )
+                    });
+                let (rhs_success_block, failure_block) =
+                    this.in_if_then_scope(local_scope, expr_span, |this| {
```

Comment:
Why do we need the additional scope here? I'd have expected a direct call to `then_else_break` to do the right thing.

Reply:
No, we don't actually need the additional scope. Thanks for the hint!